File: src/kyu_5/is_my_friend_cheating/csharp/solution.cs
using System;
using System.Collections.Generic;

public class RemovedNumbers {
	public static List<long[]> removNb(long n) {
        List<long[]> result = new List<long[]>();
        long sum = n * (n + 1) / 2;
        for (long i = 1; i <= n; i++) {
            long j = (sum - i) / (i + 1);
            if (j <= n && i * j == sum - i - j) {
                result.Add(new long[] {i, j});
            }
        }
        return result;
	}
}

File: src/kyu_5/is_my_friend_cheating/cpp/solution.cpp
#include <vector>

class RemovedNumbers
{
public:
	static std::vector<std::vector<long long>> removNb(long long n)
  {
    std::vector<std::vector<long long>> retVec;
    long long sum = (n*n+n)/2;
    for(long long i=n/2; i<=n; i++)
      if((sum-i)%(1+i)==0){
        retVec.push_back(*new std::vector<long long>{i, (sum-i)/(1+i)});
      }
    return retVec;
  }
};